### PR TITLE
Perf

### PR DESCRIPTION
**Pull request by valeriangalliat**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -9,15 +9,33 @@ const pixelIntervalPreset = { fast: 8, good: 5, best: 3, nearest: 2, bilinear: 1
 
 // Make a function to get the average color from given image data array
 // with given sample rate.
-const makeGetAverageColor = pixelInterval => data => data
-  .filter((_, index) => index % (pixelInterval * 4) < 4)
-  .reduce((rgb, channel, index) => (rgb[index % 4] += channel, rgb), [0, 0, 0])
-  .map(channel => Math.floor(channel / (data.length / 4)))
+const makeGetAverageColor = pixelInterval => data => {
+  let rgb = [0, 0, 0]
+  let len = data.length
```
Comment (KittyGiraudel) [inline lib/index.js]: No need for length caching.

File: lib/index.js [hunk 2]
```diff
@@ -9,15 +9,33 @@ const pixelIntervalPreset = { fast: 8, good: 5, best: 3, nearest: 2, bilinear: 1
...
+
+  for (let i = 0; i < len; i++) {
+    if (i % (pixelInterval * 4) < 4) {
+      rgb[i % 4] += data[i]
```
Comment (KittyGiraudel) [inline lib/index.js]: Could inline this after `if`.

Comment (valeriangalliat) [inline lib/index.js]: Linter will complain I think

File: lib/index.js (hunk 1 above)
Comment (valeriangalliat) [inline lib/index.js]: Indeed

File: lib/index.js (hunk 2 above)
Comment (KittyGiraudel) [inline lib/index.js]: I don’t think so. Not as far as I know at least.

Comment (valeriangalliat) [inline lib/index.js]: You're right. It's the `for` I can't inline.

Comment (valeriangalliat) [inline lib/index.js]: Or I can, but to be exact, I can't inline both `for` and `if` this way:

``` js
for (let i = 0; i < data.length; i++)
  if (i % (pixelInterval * 4) < 4)
    rgb[i % 4] += data[i]
```

Anyway.

**Pull request merged**